homeassistant/components/keenetic_ndms2/config_flow.py

```diff
@@ -84,10 +87,43 @@ async def async_step_user(self, user_input=None):
             errors=errors,
         )
 
-    async def async_step_import(self, user_input=None):
+    async def async_step_import(self, user_input: dict = None):
         """Import a config entry."""
         return await self.async_step_user(user_input)
 
+    async def async_step_ssdp(self, discovery_info: dict):
+        """Handle a discovered device."""
+        # Filter out items not having "keenetic' in their name
+        if (
+            "keenetic"
+            not in discovery_info.get(ssdp.ATTR_UPNP_FRIENDLY_NAME, "").lower()
+        ):
+            return self.async_abort(reason="not_keenetic_ndms2")
+
+        # filters out items having no/empty UDN
+        if not discovery_info.get(ssdp.ATTR_UPNP_UDN):
+            return self.async_abort(reason="no_udn")
+
+        host = urlparse(discovery_info[ssdp.ATTR_SSDP_LOCATION]).hostname
+        unique_id = discovery_info[ssdp.ATTR_UPNP_UDN]
+
+        if self._already_configured(host):
+            return self.async_abort(reason="already_configured")
+
+        await self.async_set_unique_id(unique_id)
```

Comment:
We should swap these, and abort if the unique ID is configured first. This is much lighter to handle/process.

Additionally, you could update the existing entry when aborting on an existing unique ID to automatically update e.g., the host of the entry if it changes.